### PR TITLE
add ut for dependencies

### PR DESCRIPTION
**Pull request by AllenZMC**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
add ut for dependencies
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
cc @Poor12 @huone1 Could you please help with this?

File: pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		},
+	}
+
+	tests := []struct {
+		name     string
+		pod      *corev1.Pod
+		expected sets.String
+	}{
+		{
+			name:     "get secret names form pod",
```
Comment (Poor12) [inline pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go:41]: from

File: pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		},
+	}
+
+	tests := []struct {
+		name     string
+		pod      *corev1.Pod
+		expected sets.String
+	}{
+		{
+			name:     "get configMap names form pod",
```
Comment (Poor12) [inline pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go:88]: from

Comment (AllenZMC) [reply pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go:88]: oh, sorry for the typo

Comment (AllenZMC) [reply pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go]: Resolved. @Poor12

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

/hold
for @Poor12 to confirm.

**Pull request merged**